fetch.bs

```diff
@@ -6102,6 +6099,21 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
  <li><p>If <var>values</var> <a for=list>contains</a> the result of
  <a>serializing a request origin</a> with <var>request</var>, then return success.
 
+ <li>
+  <p>If <var>request</var>'s <a for=request>mode</a> is "<code>navigate</code>", and
+  <var>request</var>'s <a for=request>current URL</a>'s <a for=url>origin</a> is not
+  <a>same origin</a> with <var>request</var>'s <a for=request>origin</a>, then
+  return failure.
```

Comment:
So this succeeds when A1 embeds A2. It does not succeed when A1 embeds B. It also succeeds when A1 embeds A2 and A2 redirects to B which redirects to A3. That case seems concerning.

This should probably also use "redirect-tainted origin"?

Reply:
It wouldn't succeed if `A2` redirects to `B` which redirects to `A3`, as the TAO check is done in the `B` redirect response and would set the TAO failed flag. And it uses the [serialized request origin](https://fetch.spec.whatwg.org/#serializing-a-request-origin) which takes redirect-tainting into account.

All of these concerns are baked into the TAO check already, all my patch does is ignore the `basic` [response tainting](https://fetch.spec.whatwg.org/#concept-request-response-tainting) when in a nested navigation.